electricitymap/contrib/lib/models/event_lists.py

```diff
@@ -251,6 +251,30 @@ def merge_production_breakdowns(
             production_breakdowns.events.append(prod)
         return production_breakdowns
 
+    @staticmethod
+    def update_production_breakdowns(
+        production_breakdowns: "ProductionBreakdownList",
+        new_production_breakdowns: "ProductionBreakdownList",
+        logger: Logger,
+    ) -> "ProductionBreakdownList":
+        """Given a new batch of production breakdowns, update the existing ones."""
+        if len(new_production_breakdowns) == 0:
+            return production_breakdowns
+        if len(production_breakdowns) == 0:
+            return new_production_breakdowns
+
+        for event in new_production_breakdowns.events:
```

Comment:
Isn't there a more computationally efficient way of doing this? Don't we have some guarantees on the ordering of the datetimes?

Reply:
We have a guarantee the order is correct but not that all date times are there (maybe not for the forecasts themselves) this is the most simple yet error reliant way to handle it I could think of.